src/condor_utils/submit_utils.cpp

```diff
@@ -5361,25 +5365,52 @@ int SubmitHash::SetRequirements()
 				close_op = "))";
 			}
 
+
+			// if the job is supplying transfer plugins, then we need to use a different xfer_check
+			// expression, and we want to ignore plugin types that are supplied by the job when
+			// fixing up the transfer plugin methods expression.
+			std::string xferplugs;
+			if (job->LookupString(ATTR_TRANSFER_PLUGINS, xferplugs) && ! xferplugs.empty()) {
+				xfer_check = "TARGET." ATTR_HAS_JOB_TRANSFER_PLUGINS;
+			}
+
 			answer += join_op;
 			answer += xfer_check;
+			answer += crypt_check;
+
+
+			if ( ! checks_file_transfer_plugin_methods) {
+				classad::References methods;    // methods referened by TransferInputFiles that are not in TransferPlugins
+				classad::References jobmethods; // plugin methods (like HTTP) that are supplied by the job's TransferPlugins
+
+				// xferplugs is of the form "TAR:mytarplugin; HTTP,HTTPS:myhttplugin"
+				StringTokenIterator plugs(xferplugs.c_str(), 100, ";");
+				for (const char * plug = plugs.first(); plug != NULL; plug = plugs.next()) {
+					const char * colon = strchr(plug, ':');
+					if (colon) {
+						std::string methods(plug, colon - plug);
+						add_attrs_from_string_tokens(jobmethods, methods);
+					}
+				}
 
-			if (( ! checks_file_transfer_plugin_methods)) {
-				classad::References methods;
 				// check input
 				auto_free_ptr file_list(submit_param(SUBMIT_KEY_TransferInputFiles, SUBMIT_KEY_TransferInputFilesAlt));
 				if (file_list) {
 					StringList files(file_list.ptr(), ",");
 					for (const char * file = files.first(); file; file = files.next()) {
-						if (IsUrl(file)){ methods.insert(getURLType(file)); }
+						if (IsUrl(file)){
+							MyString tag = getURLType(file);
```

Comment:
Just as a side-note, I really dislike the `IsUrl` interface given that `getURLType` internally invokes `IsUrl` as well.

Would prefer it be something with the following signature:

```c++
bool IsUrl(std::string resource, std::string *schema)
```

Something for the TODO pile.